docs/blog/posts/multi-stage-delivery-using-gitops.md

```diff
@@ -0,0 +1,409 @@
+the two `KeptnWorkloads` that are part of our `KeptnApp`.
+Let's take the `simple-go-service` `Deployment` as an example
+to see how we prepared it to be managed by Keptn:
+
+```yaml
+{% include "./multi-stage-delivery-using-gitops/deployment.yaml" %}
+```
+
+#### Labels
+
+To correctly associate the `Deployment` resource with the `KeptnWorkload` resource,
+the following labels are set:
+
+- `app.kubernetes.io/name`: The name of the `KeptnWorkload` that should be associated with the `Deployment`.
+- `app.kubernetes.io/part-of`: The name of the `KeptnApp` containing the two workloads.
+- `app.kubernetes.io/version`: The version for the related `KeptnWorkload`.
+
+#### Pre and post-deployment tasks
+
+In addition to the labels which define the `KeptnWorkload`, we also use
+the `keptn.sh/post-deployment-tasks` to define a post-deployment task for the
+workload.
+The task defined here (`wait-for-monitoring`) ensures that the Prometheus
+target for the workload is available, before proceeding with
+the execution of the load tests of the overall application.
+
+#### KeptnAppContext
+
+The `KeptnAppContext` provides two important capabilities for multi-stage delivery:
+
+- Define tasks and evaluations that run before or after the application deployment
+- Add metadata and links to traces for a specific application.
+This enables you to enrich your traces
+with additional information that you can use
+to understand and analyze the performance of your applications
+which looks as follows:
+
+  ```yaml
+  {% include "./multi-stage-delivery-using-gitops/keptnappcontext.yaml" %}
+  ```
+
+This resource contains a list of pre- and post-deployment checks
+for the complete application.
+In the `pre-deployment` phase, the task `wait-for-monitoring`
+ensures the Prometheus installation in our cluster is available.
+If this is not the case, it would not be wise to deploy a new
+version of the application, since we cannot observe the
+performance metrics of our application.
+
+Once all workloads have been deployed, the application enters the
+`post-deployment` phase, in which load tests against the application are
+executed.
+
+After executing the load tests, a `post-deployment` evaluation is
+performed, to compare the response time measured by the load tests with a threshold you have defined.
+
+If this evaluation is successful, the application proceeds into the
+`promotion` phase.
+This is the phase where the GitHub personal access token we created earlier
+is used to trigger the GitHub action to promote the deployed version
+into the next stage.
+
+#### Metadata
+
+In addition to the pre-/post-deployment checks and the promotion task,
+the `KeptnAppContext` also contains a `metadata` property that
+passes the `commitID` made available by ArgoCD to the
+application deployment.
+This information is then added by Keptn as an attribute to the
+OpenTelemetry traces created for the application deployment.
+
+To configure the application, the `values.yaml`
```

Comment:
Is this the ArgoCD Application or the KeptnApp?

Reply:
that's for configuring the helm chart which holds all the manifests from which the KeptnApp will be derived